lib/core/src/Cardano/Wallet/Primitive/Types/TokenBundle.hs

```diff
@@ -330,6 +334,14 @@ getQuantity = TokenMap.getQuantity . tokens
 hasQuantity :: TokenBundle -> AssetId -> Bool
 hasQuantity = TokenMap.hasQuantity . tokens
 
+-- | Sets the quantity associated with a given asset.
+--
+-- If the given quantity is zero, the resultant bundle will not have an entry
```

Comment:
An interface where the difference between zero quantity and "existence" smells a bit to me, but this isn't really that important.

Reply:
> An interface where the difference between zero quantity and "existence" is directly observable like this smells a bit to me, but this isn't really that important.

Definitely agree!

I would argue that this sentence (and others like it) should be removed, as:
- `TokenMap` and `TokenBundle` both aim to provide a **_total_** mapping from assets to values: it's _always_ possible to retrieve a value for a given asset.
- The fact that empty values are never stored in the internal map is an _implementation detail_.

However, for this PR I thought it best to retain consistency with the original interface.

I hope eventually to revise the API in a separate PR. (Though this may become unnecessary, once we remove `TokenMap` and `TokenBundle` and replace them with Cardano API / ledger types.)